src/CommandHook.cpp

```diff
@@ -0,0 +1,92 @@
+#include "CommandHook.hpp"
+#include "CommandRunner.hpp"
+
+#include <stout/json.hpp>
+#include <stout/protobuf.hpp>
+
+namespace criteo {
+namespace mesos {
+
+CommandHook::CommandHook(const std::string& runTaskLabelCommand,
+                         const std::string& executorEnvironmentCommand,
+                         const std::string& removeExecutorCommand)
+  : m_runTaskLabelCommand(runTaskLabelCommand),
+    m_executorEnvironmentCommand(executorEnvironmentCommand),
+    m_removeExecutorCommand(removeExecutorCommand)
+{
+}
+
+template<class Proto>
+Result<Proto> toProtobuf(const std::string& output) {
+  if(output.empty()) return Error("No content to parse");
+
+  auto outputJsonTry = JSON::parse(output);
+  if(outputJsonTry.isError()) {
+    LOG(WARNING) << "Error when parsing string to JSON.";
+    return Error("Malformed JSON");
+  }
+
+  auto outputJson = outputJsonTry.get();
```

Comment:
Would

``` 
auto outputJson = outputJsonTry.get();

if(!outputJson.is<JSON::Object>()) {
  return Error("Malformed Protobuf");  
}

auto proto = ::protobuf::parse<Proto>(outputJson);
if(proto.isError())
  LOG(WARNING) << "Error when parsing JSON to protobuf";
return proto;

``` 
 be better if you want to avoid unnecessary None() return ?

Reply:
Will do